spec/cb/utils/validator_spec.rb

```diff
@@ -12,127 +12,131 @@
+
+        it { expect { validation }.to raise_error(Cb::DocumentNotFoundError) }
+      end
+
+      context 'of 500' do
+        let(:response_code) { 500 }
+
+        it { expect { validation }.to raise_error(Cb::ServiceUnavailableError) }
+      end
+
+      context 'of 503' do
+        let(:response_code) { 503 }
+
+        it { expect { validation }.to raise_error(Cb::ServiceUnavailableError) }
+      end
+
+      context 'with valid json content' do
+        let(:response_code) { 500 }
+        let(:response_body) { '{"TestJson":{"Test":"True"}}' }
+
+        it 'returns the parsed json on the exception' do
+          expect { validation }.to raise_error do |error|
+            expect(error.response).to be_a Hash
+            expect(error.response).to have_key 'TestJson'
+            expect(error.response['TestJson']).to eq({ 'Test' => 'True' })
+          end
         end
       end
 
-      it 'when status code is 500' do
-        allow(response.response).to receive(:body).and_return('{"Errors":["Something went terribly wrong."]}')
-        allow(response).to receive(:code).and_return 500
-        expect { ResponseValidator.validate(response) }.to raise_error do |error|
-          expect(error.message).to eq '["Something went terribly wrong."]'
-          expect(error).to be_a Cb::ServerError
+      context 'with html content' do
+        let(:response_code) { 500 }
+        let(:response_body) { '<!DOCTYPE html></html>' }
+
+        it 'returns an empty hash on the exception' do
+          expect { validation }.to raise_error do |error|
+            expect(error.response).to be_empty
+          end
         end
       end
 
-      it 'when status code is 500 without errors node' do
-        allow(response.response).to receive(:body).and_return('{}')
-        allow(response).to receive(:code).and_return 500
-        expect { ResponseValidator.validate(response) }.to raise_error do |error|
-          expect(error.message).to eq ''
+      it 'populates error message from the json response' do
+        expect { validation }.to raise_error do |error|
+          expect(error.message).to eq '["Something went terribly wrong."]'
           expect(error).to be_a Cb::ServerError
         end
       end
+    end
 
-      it 'when status code is 403' do
-        allow(response.response).to receive(:body).and_return('{"errors":[]}')
-        allow(response).to receive(:code).and_return 400
-        expect { ResponseValidator.validate(response) }.to raise_error(Cb::BadRequestError)
+    context 'when there are json parsing errors' do
+      before do
+        allow(JSON).to receive(:parse).and_raise JSON::ParserError
       end
 
-      it 'when status code is 404' do
-        allow(response.response).to receive(:body).and_return('{"errors":[]}')
-        allow(response).to receive(:code).and_return 404
-        expect { ResponseValidator.validate(response) }.to raise_error(Cb::DocumentNotFoundError)
-      end
+      context 'with an XML body' do
```

Comment:
"when there are json parsing errors with an XML body" ... What? This is a thing??